cmd/dumpling/main.go

```diff
@@ -68,13 +68,11 @@ func main() {
 		fmt.Printf("\ncreate dumper failed: %s\n", err.Error())
 		os.Exit(1)
 	}
-	defer dumper.Close()
```

Comment:
can this be removed?

Reply:
The is reported by go-lint. Run `defer dumper.Close()` after `os.Exit(1)`.
I think we don't need to we dumper to `Close` when we finish dumping or meet some error. So I delete this line.

Reply:
that's strange, because in normal case (`err == nil`) the `os.Exit(1)` will not be executed and thus the `defer` would be run.

Reply:
```
cmd/dumpling/main.go:76:3: exitAfterDefer: os.Exit clutters `defer dumper.Close()` (gocritic)
                os.Exit(1)
```
This error is reported from `gocritic`.

Reply:
it looks like [the lint](https://go-critic.github.io/overview.html#exitAfterDefer-ref)'s purpose is to ensure the `defer` get executed, not removing it.

maybe change this instead to:

```go
func run() error {
    ...
    if err != nil {
        log.Error(...)
        return err
    }
    ...
}

func main() {
    if err := run(); err != nil {
        os.Exit(1)
    }
}
```